### PR TITLE
feat: adds clear button template

### LINKED ISSUE #2071: Clear button template
**Is your feature request related to a problem? Please describe.**
As a team developper, it is my duty to use my company graphical theme which includes custom icon for clear action. At the moment it is impossible for me to redefine the shape of the clear button.

**Describe the solution you'd like**
Just like the custom template feature currently available for label, option and various other parts of the `ng-select` component, I would like to add a custom template for the clear button.

### PR DESCRIPTION
**Pull request by nubinub**
resolves #2071  

Clear button template to customize clear button.

Sorry for the closed issue digging, but this is something I'd really like to have when using ng-select.

### CODE REVIEW

**Review by osamagamal65 — APPROVED** [review]

File: src/ng-select/lib/ng-select.component.html [hunk 1]
```diff
@@ -61,7 +61,13 @@
     </ng-container>
 
     <span *ngIf="showClear()" class="ng-clear-wrapper" title="{{clearAllText}}">
-        <span class="ng-clear" aria-hidden="true">×</span>
+        <ng-template #defaultClearButtonTemplate>
+            <span class="ng-clear">×</span>
+        </ng-template>
+
+        <ng-template
+            [ngTemplateOutlet]="clearButtonTemplate || defaultClearButtonTemplate">
```
Comment (pavankjadda) [inline src/ng-select/lib/ng-select.component.html:69]: Where is `clearButtonTemplate` defined?

**Review by pavankjadda — CHANGES_REQUESTED** [review]

File: src/ng-select/lib/ng-templates.directive.ts [hunk 2]
```diff
@@ -80,3 +80,8 @@ export class NgTagTemplateDirective {
 export class NgLoadingSpinnerTemplateDirective {
     constructor(public template: TemplateRef<any>) { }
 }
+
+@Directive({ selector: '[ng-clearbutton-tmp]' })
+export class NgClearButtonTemplateDirective {
+    constructor(public template: TemplateRef<any>) { }
```
Comment (pavankjadda) [inline src/ng-select/lib/ng-templates.directive.ts:86]: Why do we need this? We can pass template reference directly

**Review by pavankjadda — CHANGES_REQUESTED** [review]

**Comment by pavankjadda** [comment]
Related to #2354

**Review by pavankjadda — APPROVED** [review]
Looks good now

**Comment by pavankjadda** [comment]
@nubinub rebase with master and fix test cases

**Comment by pavankjadda** [comment]
#2387  will take care of this

**Pull request closed without merging**